src/tile/tile.ts

```diff
@@ -75,3 +82,59 @@ export const renderTile = (tile: Tile): string => {
 
   return `${dynamicChalk(tile.symbol)} `;
 };
+
+export const flatternTiles = (tiles: Tile[][]): Tile[] => {
+  const initialValue: Tile[] = [];
+  return tiles.reduce((accumulator, symbol) => {
+    return [...accumulator, ...symbol];
+  }, initialValue);
+};
+
+export const playTile = (tile: PlayableTile, player: Player): PlayableTile => {
+  return { ...tile, playedBy: player, lastPlayed: true };
+};
+
+export const findTile = (board: Board, toFind: Tile): TileCoordinate => {
+  let x: number = null;
+  let y: number = null;
+
+  board.forEach((line, l) => {
+    line.forEach((tile, t) => {
+      if (
+        tile != undefined &&
+        tile.symbol === toFind.symbol &&
+        tile.color === toFind.color
+      ) {
+        y = t;
+        x = l;
+      }
+    });
+  });
+
+  return { x, y };
+};
+
+export const removeLastPlayed = (tile: PlayableTile): PlayableTile => {
+  return { ...tile, lastPlayed: false };
+};
+
+export const findLastPLayed = (board: Board): TileCoordinate => {
+  let x: number = null;
+  let y: number = null;
+
+  board.forEach((line, l) => {
+    line.forEach((tile, t) => {
+      if (tile != undefined && tile.lastPlayed === true) {
```

Comment:
FYI, this can be simplified to

```suggestion
      if (tile?.lastPlayed) {
```

Reply:
Done